src/com/google/cose/utils/Algorithm.java

```diff
@@ -30,6 +30,8 @@ public enum Algorithm {
   SIGNING_ALGORITHM_ECDSA_SHA_384(-35, "SHA384withECDSA"),
   SIGNING_ALGORITHM_ECDSA_SHA_512(-36, "SHA512withECDSA"),
   SIGNING_ALGORITHM_EDDSA(-8, "NonewithEdDSA"),
+  SIGNING_ALGORITHM_ED25519(-9, "NonewithEd25519"),
```

Comment:
`-9` and `-10` are not valid algorithms, AFAICT.

These should match up with the IANA values: https://www.iana.org/assignments/cose/cose.xhtml#algorithms

Key size is not part of the algorithm. If we want to include key size, then we'll have to add a new field (IMO, this is the best way forward).

If you do add key size, then we'll need to remove `SIGNING_ALGORITHM_EDDSA`.